lib/ToolbarPanelHub.jsm

```diff
@@ -261,6 +280,30 @@ class _ToolbarPanelHub {
   _hideElement(document, id) {
     document.getElementById(id).setAttribute("hidden", true);
   }
+
+  _sendTelemetry(ping) {
+    this._dispatch({
+      type: "TOOLBAR_PANEL_TELEMETRY",
+      data: { action: "cfr_user_event", source: "CFR", ...ping },
+    });
+  }
+
+  sendUserEventTelemetry(win, event, message, options = {}) {
+    // Only send pings for non private browsing windows
+    if (
+      win &&
+      !PrivateBrowsingUtils.isBrowserPrivate(
+        win.ownerGlobal.gBrowser.selectedBrowser
+      )
+    ) {
+      this._sendTelemetry({
+        message_id: message.id,
+        bucket_id: message.id,
+        event,
+        value: options.value,
```

Comment:
Let's send an empty Object for `value` if `optins.value` is absent. I.e. s/`option.value`/`option.value || {}`/.

Reply:
Turns out, JSON.stringify automatically omits the properties with `undefined` values, this is no longer required.